spec/examples/by_set_pos_spec.rb

```diff
@@ -0,0 +1,103 @@
+require File.dirname(__FILE__) + '/../spec_helper'
+
+module IceCube
+  describe WeeklyRule, 'BYSETPOS' do
+    it 'should behave correctly' do
+      # Weekly on Monday, Wednesday, and Friday with the week starting on Wednesday, the last day of the set
+      schedule = IceCube::Schedule.from_ical("RRULE:FREQ=WEEKLY;COUNT=4;WKST=WE;BYDAY=MO,WE,FR;BYSETPOS=-1")
```

Comment:
For each of the frequencies, it would be nice to have the following test cases:

- With BYxxx that expands the set of recurrence instances...
  - BYSETPOS with a positive value
  - BYSETPOS with a negative value
  - BYSETPOS with multiple positive values
  - BYSETPOS with multiple negative values
  - BYSETPOS with multiple values both positive and negative
  - BYSETPOS with multiple repeated values - spec does not define behaviour; rrule.js does some weird things and outputs the same occurrence multiple times (bug or maybe just undefined behaviour?)
  - BYSETPOS with a value that matches in some intervals but not others (e.g. selecting the 5th Monday of every month since some months will have only 4 Mondays)
  - BYSETPOS with a positive value beyond than the number of recurrence instances in any interval (e.g. the 10th Monday of every month); rrule.js returns an empty set of occurrences
  - BYSETPOS with a negative value bigger than the number of recurrence instances in any set (e.g. the -10th Monday of every month); rrule.js returns a non-empty set of occurrences, which is weird (bug or maybe just undefined behaviour?)
- BYSETPOS with BYxxx that limits the set of recurrence instances instead of expanding it
- BYSETPOS with no BYxxx - spec says BYSETPOS "MUST only be used in conjunction with another BYxxx rule part" so not sure if this should be an exception or an empty set of occurrences returned; rrule.js does the latter
- BYSETPOS with multiple BYxxx rules parts that expand the recurrence instances
- BYSETPOS with multiple BYxxx rules parts that both expand and limit the recurrence instances
- BYSETPOS with a value outside the allowed range (covered in the `from_ical` tests but can you also construct a schedule using BYSETPOS without `from_ical`? if not, should that be a feature of this PR?)